rumqttd/src/server/tls.rs

```diff
@@ -227,3 +219,33 @@ impl TLSAcceptor {
         Ok(TLSAcceptor::Rustls { acceptor })
     }
 }
+
+#[cfg(feature = "use-rustls")]
+/// Get the first private key in a PEM file
+fn first_private_key_in_pemfile(key_path: &String) -> Result<PrivateKey, Error> {
+    // Get private key
+    let key_file = File::open(key_path);
+    let key_file = key_file.map_err(|_| Error::ServerKeyNotFound(key_path.clone()))?;
+
+    // Let's iterate over the keys with read_one/read_all
+    let rd = &mut BufReader::new(key_file);
+
+    loop {
+        match rustls_pemfile::read_one(rd) {
+            Ok(maybe_inner) => match maybe_inner {
+                None => {
+                    error!("No private key found in {:?}", key_path);
+                    return Err(Error::InvalidServerKey(key_path.clone()));
+                }
+                Some(Item::ECKey(key)) => return Ok(PrivateKey(key)),
+                Some(Item::RSAKey(key)) => return Ok(PrivateKey(key)),
+                Some(Item::PKCS8Key(key)) => return Ok(PrivateKey(key)),
+                Some(_) => {}
+            },
+            Err(err) => {
+                error!("Error reading key file: {:?}", err);
```

Comment:
hope this `err` doesn't flood logs unnecessarily, plz check it once!

Reply:
Not sure what you mean here because it returns after reaching the log:

```rust
            Err(err) => {
                error!("Error reading key file: {:?}", err);
                **return Err(Error::InvalidServerKey(key_path.clone()));**
            }
```

Reply:
In this line, we print the `err`

```rust
error!("Error reading key file: {:?}", err);
```

I'm not sure how long / verbose that returned error will be, so just wanted to confirm if it's not that long! Something crazy would be like printing the whole wrong key file.

If it's too long, we can avoid printing the exact err

Reply:
update: after reading docs / source, it seems it will be:

```
Error reading key file: Err(Error::InvalidData("MIIIPDCCBySgAwIBAgIQB2XGTnTlkdaAOcoqhHVj8DANBgkqhkiG9w0BAQsFADBw"))
```

Or something similar to this which has whole line. Don't think it's that long, so this should be fine. Plz correct me if i got something wrong, thanx 😁